File: src/nerd/tuxmobil/fahrplan/camp11/MyApp.java
package nerd.tuxmobil.fahrplan.camp11;

import java.util.ArrayList;

import android.app.Application;

public class MyApp extends Application {
	public static Application app = null;
	public static boolean DEBUG = true;
	public static ArrayList<Lecture> lectureList = null;
	public static int numdays;
	public static String version;
	public static String title;
	public static String subtitle;
	
	enum TASKS {
		NONE,
		FETCH,
		PARSE
	}
	
	// requestCodes für startActivityForResult
	final public static int ALARMLIST = 1;
	
	public static TASKS task_running = TASKS.NONE;
	public static String fahrplan_xml;
	public static int lectureListDay = 0;
	
    @Override
    public void onCreate() {
        super.onCreate();
        app = this;
        task_running = TASKS.NONE;
        lectureList = null;
    }
    
}

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt
package nerd.tuxmobil.fahrplan.congress

import android.app.Application
import androidx.annotation.CallSuper
import info.metadude.android.eventfahrplan.commons.logging.Logging
import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
import org.ligi.tracedroid.TraceDroid


class MyApp : Application() {

    enum class TASKS {
        NONE, FETCH, PARSE
    }

    companion object {

        @JvmField
        var taskRunning = TASKS.NONE

    }

    @CallSuper
    override fun onCreate() {
        super.onCreate()
        TraceDroid.init(this)
        taskRunning = TASKS.NONE
        AppRepository.initialize(
            context = applicationContext,
            logging = Logging.get()
        )
    }

}
